README.rst
.. warning::
  This API is deprecated.

Python Client for Game Servers API
==================================

|stable| |pypi| |versions|

`Game Servers API`_: provides access to the Game Services API.

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-game-servers.svg
   :target: https://pypi.org/project/google-cloud-game-servers/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-game-servers.svg
   :target: https://pypi.org/project/google-cloud-game-servers/
.. _Game Servers API: https://cloud.google.com/game-servers/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/gameservices/latest
.. _Product Documentation:  https://cloud.google.com/game-servers/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Game Servers API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Game Servers API.:  https://cloud.google.com/game-servers/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current `active`_ and `maintenance`_ versions of
Python.

Python >= 3.7

.. _active: https://devguide.python.org/devcycle/#in-development-main-branch
.. _maintenance: https://devguide.python.org/devcycle/#maintenance-branches

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an `end-of-life`_
version of Python, we recommend that you update as soon as possible to an actively supported version.

.. _end-of-life: https://devguide.python.org/devcycle/#end-of-life-branches

Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-game-servers


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-game-servers

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Game Servers API
   to see other available methods on the client.
-  Read the `Game Servers API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Game Servers API Product documentation:  https://cloud.google.com/game-servers/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.8.3](https://github.com/googleapis/python-game-servers/compare/v1.8.2...v1.8.3) (2023-07-04)


### Bug Fixes

* Add async context manager return types ([#413](https://github.com/googleapis/python-game-servers/issues/413)) ([96510ef](https://github.com/googleapis/python-game-servers/commit/96510eff75c247b15481b4195253230f6760f0e9))

## [1.8.2](https://github.com/googleapis/python-game-servers/compare/v1.8.1...v1.8.2) (2023-04-04)


### Documentation

* Delete samples/snippets directory ([#402](https://github.com/googleapis/python-game-servers/issues/402)) ([f17f680](https://github.com/googleapis/python-game-servers/commit/f17f680aa2f933815bf84b7510517fd4d02576b3))

## [1.8.1](https://github.com/googleapis/python-game-servers/compare/v1.8.0...v1.8.1) (2023-03-23)


### Documentation

* Fix formatting of request arg in docstring ([#399](https://github.com/googleapis/python-game-servers/issues/399)) ([ce33446](https://github.com/googleapis/python-game-servers/commit/ce3344633d1c76ee5c7fe4af59a390943fbf480e))

## [1.8.0](https://github.com/googleapis/python-game-servers/compare/v1.7.1...v1.8.0) (2023-02-28)


### Features

* Enable "rest" transport in Python for services supporting numeric enums ([cc15c8a](https://github.com/googleapis/python-game-servers/commit/cc15c8a4c15f9c408ce1b497999c4616ee1da6bc))

## [1.7.1](https://github.com/googleapis/python-game-servers/compare/v1.7.0...v1.7.1) (2023-01-20)


### Bug Fixes

* Add context manager return types ([7d5b75a](https://github.com/googleapis/python-game-servers/commit/7d5b75a3a05f306e9bf115bfd5f30bda6307006d))


### Documentation

* Add documentation for enums ([7d5b75a](https://github.com/googleapis/python-game-servers/commit/7d5b75a3a05f306e9bf115bfd5f30bda6307006d))

## [1.7.0](https://github.com/googleapis/python-game-servers/compare/v1.6.0...v1.7.0) (2023-01-10)


### Features

* Add support for python 3.11 ([#381](https://github.com/googleapis/python-game-servers/issues/381)) ([b7dae00](https://github.com/googleapis/python-game-servers/commit/b7dae00425fa279d1634b7564d95e842cec385f6))

## [1.6.0](https://github.com/googleapis/python-game-servers/compare/v1.5.3...v1.6.0) (2022-12-14)


### Features

* Add support for `google.cloud.gaming.__version__` ([2e2319f](https://github.com/googleapis/python-game-servers/commit/2e2319f997101757496d719d620d4e59469aede4))
* Add typing to proto.Message based class attributes ([2e2319f](https://github.com/googleapis/python-game-servers/commit/2e2319f997101757496d719d620d4e59469aede4))


### Bug Fixes

* Add dict typing for client_options ([2e2319f](https://github.com/googleapis/python-game-servers/commit/2e2319f997101757496d719d620d4e59469aede4))
* **deps:** Require google-api-core &gt;=1.34.0, >=2.11.0  ([575dc06](https://github.com/googleapis/python-game-servers/commit/575dc06672c3108580e899561059a19c7203a5c0))
* Drop usage of pkg_resources ([575dc06](https://github.com/googleapis/python-game-servers/commit/575dc06672c3108580e899561059a19c7203a5c0))
* Fix timeout default values ([575dc06](https://github.com/googleapis/python-game-servers/commit/575dc06672c3108580e899561059a19c7203a5c0))


### Documentation

* **samples:** Snippetgen handling of repeated enum field ([2e2319f](https://github.com/googleapis/python-game-servers/commit/2e2319f997101757496d719d620d4e59469aede4))
* **samples:** Snippetgen should call await on the operation coroutine before calling result ([575dc06](https://github.com/googleapis/python-game-servers/commit/575dc06672c3108580e899561059a19c7203a5c0))

## [1.5.3](https://github.com/googleapis/python-game-servers/compare/v1.5.2...v1.5.3) (2022-10-07)


### Bug Fixes

* **deps:** Allow protobuf 3.19.5 ([#367](https://github.com/googleapis/python-game-servers/issues/367)) ([179ad84](https://github.com/googleapis/python-game-servers/commit/179ad84a2adb23877954f386a208288dbe5259c9))

## [1.5.2](https://github.com/googleapis/python-game-servers/compare/v1.5.1...v1.5.2) (2022-09-29)


### Bug Fixes

* **deps:** Require protobuf >= 3.20.2 ([#364](https://github.com/googleapis/python-game-servers/issues/364)) ([b0ce24e](https://github.com/googleapis/python-game-servers/commit/b0ce24ea907250269246d94fbc02fc17844d3551))

## [1.5.1](https://github.com/googleapis/python-game-servers/compare/v1.5.0...v1.5.1) (2022-08-11)


### Bug Fixes

* **deps:** allow protobuf < 5.0.0 ([#345](https://github.com/googleapis/python-game-servers/issues/345)) ([8eefe86](https://github.com/googleapis/python-game-servers/commit/8eefe86d4757212c3691ed3a9b646ca9c180689d))
* **deps:** require proto-plus >= 1.22.0 ([8eefe86](https://github.com/googleapis/python-game-servers/commit/8eefe86d4757212c3691ed3a9b646ca9c180689d))

## [1.5.0](https://github.com/googleapis/python-game-servers/compare/v1.4.2...v1.5.0) (2022-07-14)


### Features

* add audience parameter ([7495784](https://github.com/googleapis/python-game-servers/commit/7495784cc56f317485514bf098598a0b8892a91b))


### Bug Fixes

* **deps:** require google-api-core>=1.32.0,>=2.8.0 ([#334](https://github.com/googleapis/python-game-servers/issues/334)) ([7495784](https://github.com/googleapis/python-game-servers/commit/7495784cc56f317485514bf098598a0b8892a91b))
* require python 3.7+ ([#336](https://github.com/googleapis/python-game-servers/issues/336)) ([730b1ab](https://github.com/googleapis/python-game-servers/commit/730b1ab9dafebb1f80cfa00288ad74ca0c821b08))

## [1.4.2](https://github.com/googleapis/python-game-servers/compare/v1.4.1...v1.4.2) (2022-06-06)


### Bug Fixes

* **deps:** require protobuf <4.0.0dev ([#325](https://github.com/googleapis/python-game-servers/issues/325)) ([e50bc8e](https://github.com/googleapis/python-game-servers/commit/e50bc8e670e33c3574830130b79413332c43314c))


### Documentation

* fix changelog header to consistent size ([#324](https://github.com/googleapis/python-game-servers/issues/324)) ([13b3053](https://github.com/googleapis/python-game-servers/commit/13b30537b429ac26dbe2e774cafc867d0fc82b68))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2023 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.7",
    "3.8",
    "3.9",
    "3.10",
    "3.11",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==22.3.0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.11"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/gaming_v1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install(
        'mypy',
        'types-requests',
        'types-protobuf'
    )
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
